src/lazy/any_encoding.rs

```diff
@@ -301,7 +422,11 @@ impl<'top> Iterator for LazyRawAnyMacroArgsIterator<'top> {
 
 /// A lazy raw reader that can decode both text and binary Ion.
 pub struct LazyRawAnyReader<'data> {
-    encoding: RawReaderKind<'data>,
+    // If the reader encounters an IVM that changes the encoding, the new encoding will be stored
+    // here until `next()` is called again, at which point the reader will be swapped out for one
+    // that can read the new encoding.
+    new_encoding: Option<IonEncoding>,
+    encoding_reader: RawReaderKind<'data>,
```

Comment:
It doesn't, but the reason is actually sort of interesting (for the right audience 😅). The reader knows to change encodings whenever it encounters an IVM _in the current encoding_. `E0 01 00 EA` and `E0 01 01 EA` are syntactically valid in both binary Ion 1.0 and 1.1. `$ion_1_0` and `$ion_1_1` are syntactically valid in both text 1.0 and text 1.1. However, if the binary reader encountered `$ion_1_0` it wouldn't recognize it and would surface an error. As a result, no additional logic was needed to prevent switching between text and binary.